code/game/objects/effects/landmarks.dm

```diff
@@ -1,95 +1,64 @@
+var/list/landmarks_list = list() //list of all landmarks
+var/list/landmarks_by_type = list() //list of landmark types associated with turf lists
+
+//Returns a list of all landmark turfs or an empty list
+//If readonly is false, it returns a COPY of the list, which can be changed without affecting the original list
+/proc/get_landmarks(input_type, readonly = TRUE)
+	var/list/L = landmarks_by_type[input_type] || list()
+	if(!readonly)
+		L = L.Copy()
+
+	return L
+
+/proc/pick_landmark(input_type, backup_type = /obj/effect/landmark/latejoin)
+	var/list/L = get_landmarks(input_type)
+	if(!L.len && (backup_type != input_type))
+		L = get_landmarks(backup_type)
+	if(!L.len)
+		return null
+
+	return pick(L)
+
 /obj/effect/landmark
 	name = "landmark"
 	icon = 'icons/mob/screen1.dmi'
 	icon_state = "x2"
 	anchored = 1
 	w_type=NOT_RECYCLABLE
 
+	var/destroy_on_creation = FALSE
+
 /obj/effect/landmark/New()
 	. = ..()
 	tag = text("landmark*[]", name)
 	invisibility = 101
 
-	switch(name)			//some of these are probably obsolete
-		if("shuttle")
-			shuttle_z = z
-			qdel(src)
-
-		if("airtunnel_stop")
-			airtunnel_stop = x
-
-		if("airtunnel_start")
-			airtunnel_start = x
-
-		if("airtunnel_bottom")
-			airtunnel_bottom = y
-
-		if("monkey")
-			monkeystart += loc
-			qdel(src)
-		if("start")
-			newplayer_start += loc
-			qdel(src)
-
-		if("wizard")
-			wizardstart += loc
-			qdel(src)
-
-		if("JoinLate")
-			latejoin += loc
-			qdel(src)
-		if("AssetJoinLate")
-			assistant_latejoin += loc
-			qdel(src)
-
-		//prisoners
-		if("prisonwarp")
-			prisonwarp += loc
-			qdel(src)
-	//	if("mazewarp")
-	//		mazewarp += loc
-		if("Holding Facility")
-			holdingfacility += loc
-		if("tdome1")
-			tdome1	+= loc
-		if("tdome2")
-			tdome2 += loc
-		if("tdomeadmin")
-			tdomeadmin	+= loc
-		if("tdomeobserve")
-			tdomeobserve += loc
-		//not prisoners
-		if("prisonsecuritywarp")
-			prisonsecuritywarp += loc
-			qdel(src)
-
-		if("blobstart")
-			blobstart += loc
-			qdel(src)
-
-		if("xeno_spawn")
-			xeno_spawn += loc
-			qdel(src)
-
-		if("endgame_exit")
-			endgame_safespawns += loc
-			qdel(src)
-		if("bluespacerift")
-			endgame_exits += loc
-			qdel(src)
+	if(!islist(landmarks_by_type[src.type]))
+		landmarks_by_type[src.type] = list()
+
+	var/list/L = landmarks_by_type[src.type]
+	L.Add(loc)
 
 	landmarks_list += src
-	return 1
+
+	if(destroy_on_creation)
+		qdel(src)
 
 /obj/effect/landmark/Destroy()
 	landmarks_list -= src
+
+	if(!destroy_on_creation)
+		var/list/L = landmarks_by_type[src.type]
+		if(L)
+			L.Remove(loc)
```

Comment:
if you don't want it destroyed on creation, why remove it from the list of landmarks?